frontend/packages/dev-console/src/components/monitoring/MonitoringPage.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Helmet } from 'react-helmet';
+import { match as RMatch } from 'react-router';
+import { Firehose, HorizontalNav, PageHeading, history } from '@console/internal/components/utils';
+import { ALL_NAMESPACES_KEY } from '@console/internal/const';
+import { ProjectModel } from '@console/internal/models';
+import { TechPreviewBadge } from '@console/shared';
+import NamespacedPage, { NamespacedPageVariants } from '../NamespacedPage';
+import ProjectListPage from '../projects/ProjectListPage';
+import MonitoringDashboard from './dashboard/MonitoringDashboard';
+import MonitoringMetrics from './metrics/MonitoringMetrics';
+
+export const MONITORING_NS_PAGE_URI = '/dev-monitoring/ns';
+export const MONITORING_ALL_NS_PAGE_URI = '/dev-monitoring/all-namespaces';
+
+export interface MonitoringPageProps {
+  match: RMatch<{
+    ns?: string;
+  }>;
+}
+
+const handleNamespaceChange = (newNamespace: string): void => {
+  const redirectURI =
+    newNamespace === ALL_NAMESPACES_KEY
+      ? `${MONITORING_ALL_NS_PAGE_URI}`
+      : `${MONITORING_NS_PAGE_URI}/${newNamespace}`;
+
+  history.push(redirectURI);
+};
+
+export const MonitoringPage: React.FC<MonitoringPageProps> = (props) => {
+  const activeNamespace = _.get(props, 'match.params.ns');
```

Comment:
and avoid `_.get` with `const activeNamespace = match.params.ns;`

Reply:
`match.params` will be an empty object when in all namespace so to avoid additional checks I've used _.get

Reply:
yeah doesn't matter much just nit and as even for `_.get` we are not passing any default value so will be undefined for all namespace and same will be for `match.params.ns`  as params is `{}`

Reply:
Avoided _.get